regex.c

```diff
@@ -82,9 +76,64 @@ bool match_regex(char *input)
 void free_regex(void)
 {
 	// Disable blocking regex checking
-	config.blockingregex = false;
 	int index;
-	for(index = 0; index < NUM_REGEX; index++)
+	for(index = 0; index < num_regex; index++)
 		if(regexconfigured[index])
 			regfree(&regex[index]);
 }
+
+void read_regex_from_file(void)
+{
+	FILE *fp;
+	char *buffer = NULL;
+	size_t size = 0;
+	int i = 0, errors = 0;
+
+	// Start timer for regex compilation analysis
+	timer_start(REGEX_TIMER);
+
+	// Get number of lines in the regex file
+	num_regex = countlines(files.regexlist);
+
+	if(num_regex < 0)
+	{
+		logg("INFO: No Regex file found");
+		return;
+	}
+
+	if((fp = fopen(files.regexlist, "r")) == NULL) {
+		logg("WARN: Cannot access Regex file");
+		return;
+	}
+
+	// Allocate memory for regex
+	regex = calloc(num_regex, sizeof(regex_t));
+	regexconfigured = calloc(num_regex, sizeof(bool));
+
+	// Search through file
+	// getline reads a string from the specified file up to either a
+	// newline character or EOF
+	while(getline(&buffer, &size, fp) != -1)
```

Comment:
This could be a for loop (the pieces of it are scattered in the function).

Reply:
My personal opinion is that `while(getline(&buffer, &size, fp) != -1)` is better readable than `for(;getline(&buffer, &size, fp) != -1;i++)` and it would only make one line (the `i++`) vanish.

Reply:
It could be
```c
for(int i = 0; getline(&buffer, &size, fp) != -1; i++)
```

Unless we have too old of a version of C.

Reply:
Yeah, that was my concern:
```
regex.c: In function ‘read_regex_from_file’:
regex.c:116:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  for(int i=0; getline(&buffer, &size, fp) != -1; i++)
```

Reply:
Okay, let's switch to C11 standard, then we can use this syntax.